src/game/client/shadow/w3dprojectedshadow.cpp

```diff
@@ -0,0 +1,1581 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Projected Shadow Code
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "w3dprojectedshadow.h"
+#include "assetmgr.h"
+#include "baseheightmap.h"
+#include "camera.h"
+#include "colmath.h"
+#include "drawable.h"
+#include "drawmodule.h"
+#include "dx8renderer.h"
+#include "dx8wrapper.h"
+#include "frustum.h"
+#include "globaldata.h"
+#include "missing.h"
+#include "object.h"
+#include "partitionmanager.h"
+#include "rendobj.h"
+#include "rinfo.h"
+#include "simpleobjectiterator.h"
+#include "terrainlogic.h"
+#include "texproject.h"
+#include "texture.h"
+#include "w3dmodeldraw.h"
+#include "worldheightmap.h"
+#include <cstring>
+
+using std::memset;
+using std::strcpy;
+using std::strlen;
+using std::strncpy;
+
+#ifdef BUILD_WITH_D3D8
+IDirect3DVertexBuffer8 *g_shadowDecalVertexBufferD3D;
+IDirect3DIndexBuffer8 *g_shadowDecalIndexBufferD3D;
+#endif
+
+int g_nShadowDecalVertsInBuf;
+int g_nShadowDecalStartBatchVertex;
+int g_nShadowDecalIndicesInBuf;
+int g_nShadowDecalStartBatchIndex;
+int g_nShadowDecalPolysInBatch;
+int g_nShadowDecalVertsInBatch;
+int g_drawStartX;
+int g_drawEdgeX;
+int g_drawStartY;
+int g_drawEdgeY;
+
+float g_heightMapScale = 0.645f;
+
+int g_shadowDecalVertexSize = 32768;
+int g_shadowDecalIndexSize = 65536;
+
+#ifndef GAME_DLL
+W3DProjectedShadowManager *g_theW3DProjectedShadowManager = nullptr;
+FrustumClass *g_shadowCameraFrustum;
+#endif
+
+W3DShadowTexture::W3DShadowTexture()
```

Comment:
init name and texture, comment as bugfix

Reply:
fixed.